test/src/unit-cppapi-profile.cc

```diff
@@ -62,6 +62,23 @@ struct ProfileCPPFx {
   }
 };
 
+struct expected_values_t {
+  std::string name = sm::RestProfile::DEFAULT_NAME;
+  std::string password = sm::RestProfile::DEFAULT_PASSWORD;
+  std::string payer_namespace = sm::RestProfile::DEFAULT_PAYER_NAMESPACE;
+  std::string token = sm::RestProfile::DEFAULT_TOKEN;
+  std::string server_address = sm::RestProfile::DEFAULT_SERVER_ADDRESS;
+  std::string username = sm::RestProfile::DEFAULT_USERNAME;
+};
+
+bool is_valid(const Profile& p, const expected_values_t& e) {
```

Comment:
Nit: `is_valid` isn't really what this is checking, it's more like `is_expected`